### PR TITLE
[EOC-632] RODO add checkbox for accepting privacy policy for google

### PR DESCRIPTION
**Pull request by alemikolo**
Sign in and Sign up flows for 'Google' users are separated. 

1. When the user creates account via Google he has to accept the privacy policy before he will be able to sign up. If they somehow omit front-end logic, the server's middleware `checkPolicyAgreement` check agreement also and block creating an account if there is no agreement.
2. If the user tries to sign in via Google there are two possibilities.
    - If the user has an account in EOC he will be logged in.
    - If the user has no account in EOC user gets feedback that he has to accept policies (checkbox appears) and after he accepts the polices he can use the same button to sign in (sign up). 

I created separate routes for sign in and sign up forms to have possibility to redirect to the proper view if something goes wrong on the server.

auth

sign-in

sign-up

### CODE REVIEW

File: client/src/modules/user/AuthBox/components/SignInForm.jsx [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+import AuthCheckbox from './AuthCheckbox';
 import { signIn } from 'modules/user/model/actions';
 import PendingButton from 'common/components/PendingButton';
 import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
 import { makeAbortablePromise, validateWith } from 'common/utils/helpers';
 import { UnauthorizedException } from 'common/exceptions/UnauthorizedException';
+import { GoogleIcon } from 'assets/images/icons';
+import Preloader, { PreloaderSize } from 'common/components/Preloader';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+
```
Comment (adamklepacz) [inline client/src/modules/user/AuthBox/components/SignInForm.jsx]: ??

File: client/src/modules/user/AuthBox/components/SignInForm.jsx [hunk 2]
```diff
@@ -61,11 +83,18 @@ class SignInForm extends PureComponent {
     );
   };
 
+  onPolicyChange = () =>
+    this.setState(({ isPolicyAccepted }) => ({
+      isPolicyAccepted: !isPolicyAccepted
+    }));
```
Comment (adamklepacz) [inline client/src/modules/user/AuthBox/components/SignInForm.jsx]: I would go for 

`acceptPolicy = () => this.setState({isPolicyAccepted: true});`

`dismissPolicy = () => this.setState({isPolicyAccepted: false});`

File: client/src/modules/user/AuthBox/components/SignInForm.jsx [hunk 3]
```diff
@@ -61,11 +83,18 @@ class SignInForm extends PureComponent {
...
+
   emailValidator = value =>
     validateWith(value => isEmail(value))('user.auth.input.email.invalid')(
       value
     );
 
+  handleLogin = () => this.setState({ pending: true });
```
Comment (adamklepacz) [inline client/src/modules/user/AuthBox/components/SignInForm.jsx]: this method sets pending to true, it is not responsible for handling login,
rename or remove and use `this.setState` where necessary

File: client/src/modules/user/AuthBox/components/SignUpForm.jsx [hunk 4]
```diff
@@ -50,6 +55,28 @@ class SignUpForm extends PureComponent {
...
 
+  componentDidMount() {
+    const {
+      match: {
+        params: { feedback }
+      }
+    } = this.props;
+    const { higherLevelErrors } = this.state;
+
+    if (feedback === 'agreement-required') {
```
Comment (adamklepacz) [inline client/src/modules/user/AuthBox/components/SignUpForm.jsx]: those feedback's strings comparison should be extracted to an enum or something

File: client/src/modules/user/AuthBox/components/SignUpForm.jsx [hunk 5]
```diff
@@ -50,6 +55,28 @@ class SignUpForm extends PureComponent {
...
+      this.setState({
+        higherLevelErrors: {
+          ...higherLevelErrors,
+          policyError: policyErrorMessageId
+        }
+      });
+    } else if (feedback === 'error') {
```
Comment (adamklepacz) [inline client/src/modules/user/AuthBox/components/SignUpForm.jsx]: and here also

File: client/src/modules/user/AuthBox/components/AuthLayout.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      </div>
+      {children}
+      <footer className="auth-layout__footer">
+        <a
+          className="auth-layout__link"
+          href={COMPANY_PAGE_URL}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          www.sandstream.pl
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/modules/user/AuthBox/components/AuthLayout.jsx]: we have only `export const COMPANY_PAGE_URL = 'https://sandstream.pl/';` but I created new one for this

File: client/src/common/utils/cookie.js [hunk 7]
```diff
@@ -3,10 +3,12 @@ const checkIfCookieSet = name => {
...
-    ? parts
-        .pop()
-        .split(';')
-        .shift()
+    ? Boolean(
+        parts
+          .pop()
+          .split(';')
+          .shift()
+      )
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/common/utils/cookie.js]: I wrote it in a different way but I left the ternary

File: client/src/common/utils/cookie.js [hunk 8]
```diff
@@ -3,10 +3,12 @@ const checkIfCookieSet = name => {
...
   return parts.length === 2
-    ? parts
-        .pop()
-        .split(';')
-        .shift()
+    ? Boolean(
+        parts
+          .pop()
+          .split(';')
+          .shift()
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/common/utils/cookie.js]: I wrote it in a different way

File: client/src/common/constants/variables.js [hunk 9]
```diff
@@ -1,6 +1,7 @@
 export const asyncPostfixes = ['FAILURE', 'SUCCESS'];
 export const COMPANY_NAME = 'Sandstream Development Sp. z o.o.';
 export const COMPANY_PAGE_URL = 'https://sandstream.pl/';
+export const COMPANY_WEBSITE = 'www.sandstream.pl/';
```
Comment (adamklepacz) [inline client/src/common/constants/variables.js]: can't we just use this one from line 3?

Comment (alemikolo) [reply client/src/common/constants/variables.js]: maybe `COMPANY_NAME = 'Sandstream Development Sp. z o.o.';` would be better. Like in app's footer?

File: server/config/auth.js [hunk 10]
```diff
@@ -146,18 +158,64 @@ const setUser = (req, resp, next) =>
...
+  return passport.authenticate(StrategyType.GOOGLE, (error, user) => {
+    if (error) {
+      return next(error);
+    }
+
+    if (!user) {
+      return resp.redirect(`${sourceRoute}/${RouteParams.ERROR}`);
+    }
+
+    req.login(user, error => {
```
Comment (marekrozmus) [inline server/config/auth.js]: => error ? next(error) : resp.redirect('/')

**Pull request merged**